### DISCUSSION TITLE
rtx -> mise rename

### DISCUSSION THREAD

**Discussion by jdx** [jdx/mise#1338] [Announcements]
If you're having trouble with the migration, first read this doc: https://mise.jdx.dev/rtx.html

I also welcome you to our discord: https://discord.gg/UBa7pJUN7Z and as always, feel free to post an issue with any problems. I'll continue updating that doc as we discover migration-related issues worth noting.

I really apologize for inconvenience here. There just wasn't a way to do this that wasn't sudden I could think of.

**Comment by jdx** [comment]
Several bugs were patched today so please make sure you're on an updated version.

Note that we moved the default location `install.sh` uses from `~/.local/share/rtx/bin/rtx` to `~/.local/bin/mise` to keep things cleaner and reuse a common convention. This technically happened post-migration so some people may be using `~/.local/share/mise/bin/mise`. That shouldn't cause any problems. It doesn't really matter where the CLI is. `mise self-update` will just use the location it's currently in. I figured I should mention this but it should only impact new installs. Docs reflect the new location already.

**Comment by zyrikby** [comment]
I've tried to find out but could not do this: what is the reason behind the renaming? Is it due to the new functionality appeared recently?

**Reply by sassdavid** [reply]
As I understand it mostly because of the name clash with Nvidia, it was difficult to search etc.

See: https://github.com/jdx/mise/discussions/1306

**Comment by blackhaj** [comment]
Not so much a bug as a potential conflict others might run into. Sharing the solution so they don't lose any time on it

After switching from rtx to mise, when I open a new terminal window I get the following error:
`(eval):cd:39: too many arguments`

![CleanShot 2024-01-04 at 15 28 24@2x](https://github.com/jdx/mise/assets/15220184/24e4f1e1-191c-450d-8ac2-6ce1f5417a68)

I have managed to locate the issue which is an alias I had declared that I am guessing conflicts with an alias or exposed command in mise, namely `test`:

This is my alias
`alias test="cd ~/path/to/somewhere/test-suite"`

If I remove this, the issue goes away.

**To Reproduce**
Steps to reproduce the behavior.

1. Add an alias for `test` to one of your startup dotfiles e.g. .zshrc `alias test="cd ~/path/to/somewhere/test-suite"` (add it before the eval line to activate mise)
2. Open a new terminal window and you should see a similar error

**Reply by jdx** [reply]
yeah what's happening is you're creating a function named `test` so when I later run `test -n "$(declare -f command_not_found_handler)"` it doesn't use the builtin `test`.

I think it's possible to rewrite that code using `[` which would be less likely to be overridden.

**Reply by jdx** [reply]
this isn't related to the mise rename btw, it's the somewhat recent not_found handler

**Reply by blackhaj** [reply]
Ok great, thanks for the clarity. It's not an issue for me as the test alias was redundant anyway so I have deleted it but I am glad I was on the right track with my debugging.

Thanks for the great tool 🙌

**Reply by jdx** [reply]
yeah I wouldn't recommend adding an alias over a builtin like that. Any code that you "source", "eval" or paste in your terminal that uses "test" would break. Still, this is something I can easily fix. I think someone else had this problem but wasn't able to diagnose it.

**Comment by chadlwilson** [comment]
Sadly upgrading via Brew from `2024.0.0` to `2024.1.4` did not migrate installs/plugins/config for me as noted at https://mise.jdx.dev/rtx.html

It seemed to basically start fresh. After correcting my oh-my-zsh config to `eval "$(/opt/homebrew/bin/mise activate zsh)"` at the new location and running `mise` once to try and trigger the migration, I ended up manually moving the contents of `installs` (minus python) and `plugins` from the `.rtx` folders to the `.mise` folders.

Note sure if subsequently addressed or whether any help could be given to figure out why the auto-migration didn't work?

**Reply by chadlwilson** [reply]
Sadly I didn't capture the first run of `mise` after the update since it was shell initiated, nor run with higher verbosity, but I suppose I could try and replicate the issue with a rollback if it's not known.

```
$ brew config
HOMEBREW_VERSION: 4.2.2
ORIGIN: https://github.com/Homebrew/brew
HEAD: 705d2564ddfb9c915ee9425dd9ed408bfac61005
Last commit: 3 days ago
Core tap JSON: 05 Jan 02:52 UTC
Core cask tap JSON: 05 Jan 02:52 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_AUTO_UPDATE_SECS: 50400
HOMEBREW_CASK_OPTS: []
HOMEBREW_MAKE_JOBS: 10
HOMEBREW_NO_ENV_HINTS: set
Homebrew Ruby: 3.1.4 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.1.4/bin/ruby
CPU: 10-core 64-bit arm_firestorm_icestorm
Clang: 15.0.0 build 1500
Git: 2.43.0 => /opt/homebrew/bin/git
Curl: 8.4.0 => /usr/bin/curl
macOS: 14.2.1-arm64
CLT: 15.1.0.0.1.1700200546
Xcode: 15.1
Rosetta 2: false
```

**Reply by jdx** [reply]
I'm not really sure, but it might be worth at least a glance at the code (migrate_rtx()) if you might have an idea where the logic is falling over

https://github.com/jdx/mise/blob/main/src/migrate.rs

**Reply by chadlwilson** [reply]
Not directly related but I infer from the below that its not "safe" to migrate Ruby installs either? https://mise.jdx.dev/rtx.html only mentions Python so I manually moved everything except Python, but I can PR to the docs on that too.

https://github.com/jdx/mise/blob/49027919665af8a568c60356f8022d757a26e68e/src/migrate.rs#L67-L69

**Reply by jdx** [reply]
someone mentioned that and I suspect it's probably accurate, but I have not tested it to be certain myself. You're right I forgot to update the docs on that one though.

**Reply by chadlwilson** [reply]
Mine do not seem to be working, so you're probably right. I'll PR the docs to make them consistent with the impl.

Note the dylib reference to old `rtx` install location from new `bin`.

```
dyld[85289]: Library not loaded: /Users/chad/.local/share/rtx/installs/ruby/3.1.4/lib/libruby.3.1.dylib
  Referenced from: <CA8F653C-3665-3821-BB09-6FD8F7C8D6DE> /Users/chad/.local/share/mise/installs/ruby/3.1.4/bin/ruby
  Reason: tried: '/Users/chad/.local/share/rtx/installs/ruby/3.1.4/lib/libruby.3.1.dylib' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/chad/.local/share/rtx/installs/ruby/3.1.4/lib/libruby.3.1.dylib' (no such file), '/Users/chad/.local/share/rtx/installs/ruby/3.1.4/lib/libruby.3.1.dylib' (no such file), '/usr/local/lib/libruby.3.1.dylib' (no such file), '/usr/lib/libruby.3.1.dylib' (no such file, not in dyld cache)
  ```

**Comment by EamonKeane** [comment]
This one confused me for a bit, with psycopg2 referencing an rtx install `~/.local/share/rtx/installs/postgres/15.5/lib/libpq.5.dylib`. It was fixed by doing a `pip cache purge` to clear `~/Library/Caches/pip `
